src/engine/index.ts

```diff
@@ -110,27 +111,26 @@ class Engine {
   }
 
   /** Merge a cast into the set */
-  mergeCast(cast: Cast): Result<void, string> {
+  async mergeCast(cast: Cast): Promise<Result<void, string>> {
     try {
       const username = cast.data.username;
 
       const signerChanges = this._users.get(username);
       if (!signerChanges) {
         return err('mergeCast: unknown user');
       }
-
-      if (!this.validateMessage(cast).isOk()) {
-        return this.validateMessage(cast);
+      if (!(await this.validateMessage(cast)).isOk()) {
+        const validation = await this.validateMessage(cast);
+        return validation;
       }
-
       let castSet = this._casts.get(username);
       if (!castSet) {
         castSet = new CastSet();
         this._casts.set(username, castSet);
       }
-
       return castSet.merge(cast);
     } catch (e: any) {
+      console.error(e);
```

Comment:
if you want to keep log lines because they are useful for debugging, i would add some method context so its easier for the reader to see. something like: 

```suggestion
      console.error(`engine.MergeCast` + e.toString());
```